airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -71,34 +83,68 @@ protected ChangeEvent<String, String> computeNext() {
     while (!MoreBooleans.isTruthy(publisherStatusSupplier.get()) || !queue.isEmpty()) {
       final ChangeEvent<String, String> next;
       try {
-        final Duration waitTime = receivedFirstRecord ? SUBSEQUENT_RECORD_WAIT_TIME : firstRecordWaitTime;
+        // #18987: waitTime here is largely meaningless with heartbeats but leaving for backward
```

Comment:
nit : For the comment, wait time is still relevant though right? Based on your comment below where DBZ can be stuck searching for WAL resume position and timeout provides a mechanism to timeout

Reply:
Clarified comment.